### DISCUSSION TITLE
Como identificar se o xml incompleto já foi manifestado Ciência ou não?

### DISCUSSION THREAD

**Discussion by EduardoFerreira22** [TadaSoftware/PyNFe#347] [General]
## Como identificar se o xml incompleto já foi manifestado Ciência ou não? 

Como identificar se o xml que veio incompleto, já está manifestado ou não?

Utilizei o código:
`
   while True:
        xml = con.consulta_distribuicao(cnpj=CPFCNPJ, chave=CHAVE, nsu=NSU)
        print(f'Nova consulta a partir do NSU: {str(NSU).zfill(15)}')

        try:
            resposta = etree.fromstring(xml.text.encode('utf-8'))
        except etree.XMLSyntaxError as e:
            print("Erro de sintaxe XML na resposta:", e)
            print("Resposta recebida:", xml.text)
            break

        ns = {'ns': NAMESPACE_NFE}
        
        contador_resposta = len(resposta.xpath('//ns:retDistDFeInt/ns:loteDistDFeInt/ns:docZip', namespaces=ns))
        print(f'Quantidade de NSUs na consulta atual: {contador_resposta}')
        
        try:
            cStat = resposta.xpath('//ns:retDistDFeInt/ns:cStat', namespaces=ns)[0].text
            print(f'cStat: {cStat}')
        except IndexError:
            print("Elemento cStat não encontrado na resposta")
            print("Resposta recebida:", etree.tostring(resposta, pretty_print=True).decode())
            break

        xMotivo = resposta.xpath('//ns:retDistDFeInt/ns:xMotivo', namespaces=ns)[0].text
        print(f'xMotivo: {xMotivo}')    
        
        maxNSU = resposta.xpath('//ns:retDistDFeInt/ns:maxNSU', namespaces=ns)[0].text
        print(f'maxNSU: {maxNSU}')
        
        if cStat == '138':
            for contador_xml in range(contador_resposta):
                tipo_schema = resposta.xpath('//ns:retDistDFeInt/ns:loteDistDFeInt/ns:docZip/@schema', namespaces=ns)[contador_xml]
                
                # if tipo_schema == 'procNFe_v4.00.xsd': 
                #     zip_resposta = resposta.xpath('//ns:retDistDFeInt/ns:loteDistDFeInt/ns:docZip', namespaces=ns)[contador_xml].text
                #     resposta_descompactado = DescompactaGzip.descompacta(zip_resposta)
                #     texto_descompactado = etree.tostring(resposta_descompactado).decode('utf-8')

                #     chave_acesso = resposta_descompactado.xpath('//ns:NFe/ns:infNFe/@Id', namespaces=ns)[0][3:]
                #     n_nfe = resposta_descompactado.xpath('//ns:ide/ns:nNF', namespaces=ns)[0].text
                #     emitente = resposta_descompactado.xpath('//ns:emit/ns:xNome', namespaces=ns)[0].text
                #     cnpj_emitente = resposta_descompactado.xpath('//ns:emit/ns:CNPJ', namespaces=ns)[0].text
                #     destinatario = resposta_descompactado.xpath('//ns:dest/ns:xNome', namespaces=ns)[0].text
                #     cnpj_destinatario = resposta_descompactado.xpath('//ns:dest/ns:CNPJ', namespaces=ns)[0].text
                #     data_emissao = resposta_descompactado.xpath('//ns:ide/ns:dhEmi', namespaces=ns)[0].text
                #     valor_total = resposta_descompactado.xpath('//ns:total/ns:ICMSTot/ns:vNF', namespaces=ns)[0].text
                #     with open(f'./{chave_acesso}.xml', 'w+', encoding='UTF-8') as f:
                #         f.write(texto_descompactado)
                
                # baixar o resumo e realizar a manifestacao do destinatario
                if (tipo_schema == 'resNFe_v1.01.xsd'):
                    try:
                        zip_resposta = resposta.xpath('//ns:retDistDFeInt/ns:loteDistDFeInt/ns:docZip', namespaces=ns)[contador_xml].text

                        # XML completo do resumo
                        resposta_descompactado = DescompactaGzip.descompacta(zip_resposta)
                        texto_descompactado = etree.tostring(resposta_descompactado).decode('utf-8')
                        chave_acesso_nfe = resposta_descompactado.xpath('//ns:resNFe/ns:chNFe', namespaces=ns)[0].text
                        with open(f'./{chave_acesso_nfe}.xml', 'w+', encoding='UTF-8') as f:
                            f.write(texto_descompactado)
                        print(f"{resposta_descompactado}\n{texto_descompactado}")
                        # Ler chave de acesso do resumo
                        
                        
                        print(f"{chave_acesso_nfe}")
                    except Exception as e:
                        print(e)

            NSU = int(resposta.xpath('//ns:retDistDFeInt/ns:ultNSU', namespaces=ns)[0].text)
            print(f'NSU: {NSU}')
            
        elif cStat == '137':
            print('Não há mais documentos a pesquisar')
            break
        else:
            print('Falha')
            break

    # Fechar a conexão ao banco de dados
    # conn.close()

buscar_sefaz()
`

**Comment by leogregianin** [comment]
@EduardoFerreira22 não tem como saber se um XML já foi manifestado ou não. Uma opção é você fazer uma nova manifestação e visualizar qual o tipo de retorno. Lembrando que você pode fazer mais de uma manifestação para uma mesma nota, por exemplo, você pode fazer uma ciência da operação e algum tempo depois para fazer uma manifestação de confirmação ou desconhecimento.
